rpc/client/client.go

```diff
@@ -692,31 +695,56 @@ func (c *Client) BlockSearch(ctx context.Context, query string, page, perPage *i
 func (c *Client) Status(ctx context.Context) (*ctypes.ResultStatus, error) {
 	latest, err := c.node.Store.LoadBlock(c.node.Store.Height())
 	if err != nil {
-		// TODO(tzdybal): extract error
 		return nil, fmt.Errorf("failed to find latest block: %w", err)
 	}
 
-	latestBlockHash := latest.Header.DataHash
-	latestAppHash := latest.Header.AppHash
-	latestHeight := latest.Header.Height
-	latestBlockTimeNano := latest.Header.Time
+	initial, err := c.node.Store.LoadBlock(uint64(c.node.GetGenesis().InitialHeight))
+	if err != nil {
+		return nil, fmt.Errorf("failed to find earliest block: %w", err)
+	}
+
+	validators, err := c.node.Store.LoadValidators(latest.Header.Height)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch the validator info at latest block: %w", err)
+	}
+	_, validator := validators.GetByAddress(latest.Header.ProposerAddress)
+
+	defaultProtocolVersion := corep2p.NewProtocolVersion(
+		version.P2PProtocol,
+		version.BlockProtocol,
+		0,
```

Comment:
This should be:
```suggestion
		state.Version.Consensus.Block,
		state.Version.Consensus.App,
```

where `state, err := c.node.Store.LoadState()`